examples/06-ip-prefixes/adapter_ipam_a.py

```diff
@@ -0,0 +1,94 @@
+"""IPAM A adapter."""
+import os
+import ipaddress
+import yaml
+from models import Prefix  # pylint: disable=no-name-in-module
+from diffsync import DiffSync
+
+dirname = os.path.dirname(os.path.realpath(__file__))
+
+
+class IpamAPrefix(Prefix):
+    """Implementation of Prefix create/update/delete methods for IPAM A."""
+
+    @classmethod
+    def create(cls, diffsync, ids, attrs):
+        """Create a Prefix record in IPAM A."""
+        data = diffsync.load_yaml_data()
+
+        data.append(
+            {
+                "cidr": ids["prefix"],
+                "family": ipaddress.ip_address(ids["prefix"].split("/")[0]).version,
+                "vrf": attrs["vrf"],
+                "vlan": f'VLAN{attrs["vlan_id"]}',
+                "customer_id": attrs["tenant"] if attrs["tenant"] else None,
+            }
+        )
+
+        diffsync.write_yaml_data(data)
```

Comment:
Seems wrong to me to re-load and re-dump the YAML file after each individual model create/update/delete. Should this be done (once) in a `sync_complete` callback instead?

Reply:
you are totally right